coriolis/tests/api/v1/test_migrations.py

```diff
@@ -0,0 +1,139 @@
+    @mock.patch('coriolis.api.v1.migrations.CONF')
+    def test_show(
+        self,
+        mock_CONF,
+        mock_get_migrations_policy_label,
+        mock_get_migration,
+        mock_single
+    ):
+        mock_req = mock.Mock()
+        mock_context = mock.Mock()
+        mock_req.environ = {'coriolis.context': mock_context}
+        id = mock.sentinel.id
+        mock_CONF.api.include_task_info_in_migrations_api = False
+
+        result = self.migrations.show(mock_req, id)
+
+        self.assertEqual(
+            mock_single.return_value,
+            result
+        )
+
+        mock_get_migrations_policy_label.assert_called_once_with("show")
+        mock_get_migration.assert_called_once_with(
+            mock_context, id, include_task_info=False
+        )
+        mock_single.assert_called_once_with(mock_get_migration.return_value)
+
+    @ddt.data(
+        (
+            {
+                "migration": {
+                    "user_scripts": {"mock_user_scripts": None},
+                    "instances": ["mock_instance1", "mock_instance2"],
+                    "replica_id": "mock_replica_id",
+                    "clone_disks": True,
+                    "force": False,
+                    "skip_os_morphing": False,
+                    "instance_osmorphing_minion_pool_mappings":
+                        {"mock_mapping": "mock_value"},
+                }
+            }
+        ),
+        (
+            {
+                "migration": {
+                    "user_scripts": {"mock_user_scripts": None},
+                    "instances": ["mock_instance1", "mock_instance2"],
+                    "replica_id": None,
+                    "clone_disks": True,
+                    "force": False,
+                    "skip_os_morphing": False,
+                    "instance_osmorphing_minion_pool_mappings":
+                        {"mock_mapping": "mock_value"},
+                }
+            }
+        )
+    )
+    @mock.patch.object(migration_view, 'single')
+    @mock.patch.object(api.API, 'migrate_instances')
+    @mock.patch.object(migrations.MigrationController,
+                       '_validate_migration_input')
+    @mock.patch.object(api.API, 'deploy_replica_instances')
+    @mock.patch.object(api_utils, 'normalize_user_scripts')
+    @mock.patch.object(api_utils, 'validate_user_scripts')
+    @mock.patch.object(migration_policies, 'get_migrations_policy_label')
+    def test_create(
+        self,
+        data,
+        mock_get_migrations_policy_label,
+        mock_validate_user_scripts,
+        mock_normalize_user_scripts,
+        mock_deploy_replica_instances,
+        mock__validate_migration_input,
+        mock_migrate_instances,
+        mock_single
+    ):
+        mock_req = mock.Mock()
+        mock_context = mock.Mock()
+        mock_req.environ = {'coriolis.context': mock_context}
+        mock_values = [(f'mock_value_{i}') for i in range(1, 15)]
```

Comment:
Yeah.. let's not do this please. If you REALLY want to check these, use mock sentinels instead. IMO you can skip these long checks, the only thing that really matters is that the right API method is called if 'replica_id' is in `migration` or not. You can also check `validate_user_scripts` and `normalize_user_scripts` calls since you have to mock them.